docs/v1-proposed-api.md

```diff
@@ -0,0 +1,245 @@
+ 1. Move PlatformSecrets into Platform, rename Credentials to CredentialsSecret.
+ 1. Add support for bare metal provisioning.
+   * At this point, we do not see any major implications for BareMetal. This is modelled as a separate platform in install config (see [this example](https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md#install-config))
+   * This would be specified in the InstallConfig.
+   * Hive would likely want a platform indicator explicitly for bare metal if we need a special invocation of the installer.
+   * However for the most part, we do not see any major complications adding bare metal support to the Hive API.
+ 1. Add support for hosted control plane provisioning. (control plane running in Hive management cluster, in the same namespace as the ClusterDeployment, but cluster workers in a remote cloud provider account)
+   * A lot needs to be settled here before we can know exactly how this would look. However looking over the [current inputs for the POC code](https://github.com/sjenning/hosted-cluster-poc/blob/master/config.sh.aws_example), we will likely combine a Spec.Platform as we have today, with another Spec.HostedControlPlane section containing a variety of settings.
+   * May deploy control plane to the same cluster as Hive, or give Hive a kubeconfig for a remote cluster where the control plane should run.
+   * Safer to assume we are not the cluster where control plane will run.
+   * Spec.Platform.AWS specified, as well as Spec.HostedControlPlane.Kubeconfig etc.
+
+
+### Proposed V1 ClusterDeployment
+
+```yaml
+apiVersion: hive.openshift.io/v1alpha1
+kind: ClusterDeployment
+metadata:
+  name: ci-cluster-v4-1
+  namespace: uhc-staging-18urm2sbab85d9crlk27is39n706ph3g
+spec:
+  baseDomain: r4f1.s1.devshift.org
+  certificateBundles:
+  - generate: true
+    name: primary-cert-bundle
+    certificateSecret:
+      name: primary-cert-bundle-secret
+  compute: # controlPlane is gone, install config only, but may reappear here if these fall under machine set management in future.
+  - name: worker
+    platform:
+      aws:
+        rootVolume:
+          iops: 100
+          size: 300
+          type: gp2
+        type: m5.xlarge
+        zones:
+        - us-east-1a
+    replicas: 4
+  controlPlaneConfig:
+    servingCertificates:
+      default: primary-cert-bundle
+  imageSet: # immutable
+    name: openshift-v4.1.18
+  images: {} # immutable
+  ingress:
+  - domain: apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+    name: default
+    servingCertificate: primary-cert-bundle
+  clusterMetadata:
+    # Mutable until installed=true, after which we will not allow updates.
+    infraID: ci-cluster-v4-1-64bns
+    clusterID: 1e4f00a3-a30e-46d7-9567-4bb61caf6d73
+    adminKubeconfigSecret:
+      name: ci-cluster-v4-1-0-ndv7l-admin-kubeconfig
+    adminPasswordSecret:
+      name: ci-cluster-v4-1-0-ndv7l-admin-password
+  installed: true
```

Comment:
what is this field?

Reply:
It specifies whether the cluster is installed. It is set to true by the clusterdeployment controller when the installation of the cluster completes successfully. It also allows a user to create a clusterdeployment that has already been installed which they would like to have adopted by Hive.